.travis.yml

```diff
@@ -0,0 +1,27 @@
+language: rust
+sudo: false
+env:
+  global:
+    - RUST_BACKTRACE=1
+    - RUSTFMT_VERSION=0.3.4
+cache: cargo
+
+matrix:
+  include:
+    - rust: stable
+      script:
+        - cargo test --all -- --nocapture
+        - cargo test --all --benches benches -- --nocapture
+    # rustfmt and clippy (behind the 'dev' feature) only work on the nightly toolchain.
+    # This build is considerably more thorough.
+    - rust: nightly
+      allow_failures: true
+      before_script: |
+        export PATH="$PATH:$HOME/.cargo/bin"
+        if [[ -n "`which cargo-fmt`" ]] && [[ "`cargo fmt --version`" != *"$RUSTFMT_VERSION"* ]]; then
```

Comment:
What if rustfmt in cache doesn't work with latest rustc?

Reply:
I think that if the `cargo-fmt` exists and the version is correct, in this case, it will already be compiled and not depend on the latest nightly.

If `rustc` stops being able to compile this version of `rustfmt` for some reason then we would need to bump `$RUSTFMT_VERSION`.

I think this is acceptable given that we target stable and are going to allow failures on the nightly build.